boa/src/syntax/lexer/number.rs

```diff
@@ -287,42 +287,47 @@ impl<R> Tokenizer<R> for NumberLiteral {
 
         let num = match kind {
             NumericKind::BigInt(base) => {
-                Numeric::BigInt(
-                    BigInt::from_string_radix(&buf, base).expect("Could not convert to BigInt")
-                    )
+                let val = BigInt::from_string_radix(&buf, base);
+                match val {
+                    Some(bigint) => Ok(Numeric::BigInt(bigint)),
+                    _ => Err("Could not convert to BigInt"),
```

Comment:
We should not be changing this to return `Err` if it reached this point and it gave an error from `from_string_radix` that means there is a bug it the logic on top.

Here is the problem with the input `0xn` is it sets it as hexadecimal number kind, so we reach the
```
  ⬇️ 
0xn
```
and we reach the `n` we and in line `230` we consume all the number characters, and the problem is here
we need to check that at least there is one digit that has been consumed. when we are in non-decaimal mode (in base 16, 8, 2).

This error is generated also with `0on` (octal) and `0bn` (binary)

Reply:
Thank you @HalidOdat!
Could you guide me how to solve this?

Reply:
> Thank you @HalidOdat!
> Could you guide me how to solve this?

We need to peek at the char before line `230` where we start consuming the digit chars: we need to check if we are in not decimal mode, if so we peek the next char if it is not a digit according to the base that its in (16, 8, 2) and if that is the case we generate an `Err`, if not we just fall through.

Reply:
if im not mistaken the cursor has a `.peek()` method, and we can get the base with `kind.base()`, and check if its a digit with that base with `char::is_digit(base)`